### PR TITLE
Set up alternative pipeline for snippet generation

### PR DESCRIPTION
**Pull request by zengin**
- Creates a weekly snippet generation pipeline
- Replaces https://github.com/microsoftgraph/microsoft-graph-explorer-api/blob/dev/pipelines/generateSnippets.yml
  - Removal of original snippet generation will happen in a follow-up PR (#340)
- Updated branch reference from `dev` to `master` to use production version of the snippet generation tool
- Follows the same patterns as SDK generation pipelines for git operations
- Steps:
  1. Builds snippet generation tool
  2. Builds apidoctor
  3. Using these two tools and the docs input, generates snippets for the languages specified
  4. Pushes generated snippets in a specific branch format so that a Github action can be triggered for PR generation in the docs repo.

[AB#6093](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/6093)
[AB#6094](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/6094)

### CODE REVIEW

File: pipelines/snippets.yml [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Pipeline for Snippet injection into docs
+
+trigger: none
+pr: none
+schedules:
```
Comment (zengin) [inline pipelines/snippets.yml:8]: schedule is taken from the original pipeline: https://github.com/microsoftgraph/microsoft-graph-explorer-api/blob/dev/pipelines/generateSnippets.yml#L5-11

File: pipelines/snippets.yml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  - cron: "30 12 * * 2"
+    displayName: Weekly Tuesday snippets updates
+    branches:
+      include:
+      - master
```
Comment (zengin) [inline pipelines/snippets.yml:13]: branch is changed from `dev` to `master`

File: pipelines/snippets.yml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+   - repository: microsoft-graph-docs
+     type: github
+     endpoint: microsoftgraph
+     name: microsoftgraph/microsoft-graph-docs
+     ref: master
+   - repository: apidoctor
+     type: github
+     endpoint: microsoftgraph
+     name: microsoftgraph/apidoctor
+     ref: zengin/snippets
```
Comment (zengin) [inline pipelines/snippets.yml:27]: code is merged
```suggestion
     ref: generate-snippets
```

File: pipelines/snippets.yml [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+   - repository: microsoft-graph-docs
+     type: github
+     endpoint: microsoftgraph
+     name: microsoftgraph/microsoft-graph-docs
+     ref: master
+   - repository: apidoctor
+     type: github
+     endpoint: microsoftgraph
+     name: microsoftgraph/apidoctor
+     ref: generate-snippets
```
Comment (MIchaelMainer) [inline pipelines/snippets.yml:27]: Will the ref  change from `generate-snippets`? Or is the branch that snippets are always generated from?

**Review by MIchaelMainer — APPROVED** [review]

File: pipelines/snippets.yml (hunk 4 above)
Comment (zengin) [reply pipelines/snippets.yml:27]: This is the branch currently in use for the pipelines.

My improvements were based on that branch. I will defer to @andrueastman to merge that to `master`. Even if we don't merge this into `master` of `OneDrive/apidoctor`, I think it makes sense to merge it into `microsoftgraph/apidoctor`.

**Pull request merged**